README.md

```diff
@@ -394,10 +396,57 @@ The `extract` method takes no arguments:
     1. `v` must have the same type that `f` returns in `extend`.
 
 
+### Bifunctor
 
+A value that implements the Bifunctor specification is a functor with two
+type arguments that are both covariant.
 
+1. `p.bimap(a => a, b => b)` is equivalent to `p` (identity)
+2. `p.bimap(compose(f1)(f2), compose(g1)(g2))` is equivalent to `p.bimap(f1, g1).bimap(f2, g2)` (composition)
 
+#### `bimap` method
 
+A value which has a Bifunctor must provide an `bimap` method. The `bimap`
+method takes two arguments:
+
+    c.bimap(f, g)
+
+1. `f` must be a function which returns a value
+
+    1. If `f` is not a function, the behaviour of `bimap` is unspecified.
+
+2. `g` must be a function which returns a value
+
+    1. If `g` is not a function, the behaviour of `bimap` is unspecified.
+
+
+### Profunctor
+
+A value that implements the Profunctor specification must also implement
+the Functor specification. The Profunctor specification is a contravariant
```

Comment:
I think this difference is expressed as the difference in "composition" law of two types.

Btw I also think that it would be better to use type signatures in the spec instead of writing down types in english, how it's currently done.